app/scripts/controllers/membership.js

```diff
@@ -0,0 +1,304 @@
+      $scope,
+      $timeout,
+      $uibModal,
+      AuthorizationService,
+      DataService,
+      ProjectsService,
+      MembershipService,
+      RoleBindingsService,
+      RolesService,
+      LocalStorageUserStore) {
+
+      var requestContext;
+      var user = LocalStorageUserStore.getUser();
+      var projectName = $routeParams.project;
+      var startCase = $filter('startCase');
+      var formatKind = function(kindName) {
+        return startCase(kindName).toLowerCase();
+      };
+      var allRoles = [];
+
+      // NOTE: these could all be moved out into a strings service.
+      var messages = {
+        errorReason: _.template('Reason: "<%= httpErr %>"'),
+        notice: {
+          yourLastRole: _.template('Removing the role "<%= roleName %>" may completely remove your ability to see this project.')
+        },
+        warning: {
+          serviceAccount: _.template('Removing a system role granted to a service account may cause unexpected behavior.')
+        },
+        remove: {
+          areYouSure: {
+            subject: _.template('Are you sure you want to remove <strong><%= roleName %></strong> from the <%= kindName %> <strong><%= subjectName %></strong>?'),
+            self: _.template('Are you sure you want to remove <strong><%= roleName %></strong> from <strong><%= subjectName %></strong>?')
+          },
+          success: _.template('The role "<%= roleName %>" was removed from "<%= subjectName %>".'),
+          error: _.template('The role "<%= roleName %>" was not removed from "<%= subjectName %>".')
+        },
+        update: {
+          subject: {
+            success: _.template('The role "<%= roleName %>" was given to "<%= subjectName %>"'),
+            error: _.template('The role "<%= roleName %>" was not given to "<%= subjectName %>"')
+          }
+        }
+      };
+
+      // NOTE: alert service?
+      var showAlert = function(name, type, msg, detail, scope) {
+        scope = scope || $scope;
+        scope.alerts[name] = {
+          type: type,
+          message: msg
+        };
+        if(detail) {
+          scope.alerts[name].details = detail;
+        }
+      };
+
+      var resetForm = function() {
+        $scope.disableAddForm = false;
+        $scope.newBinding.name = '';
+        $scope.newBinding.namespace = null;
+        $scope.newBinding.role = '';
+      };
+
+      var refreshRoleBindingList = function() {
+        RoleBindingsService
+          .list(requestContext, function(resp) {
+            angular.extend($scope, {
+              canShowRoles: true,
+              roleBindings: resp.by('metadata.name'),
+              subjectKindsForUI: MembershipService.mapRolebindingsForUI(resp.by('metadata.name'), allRoles)
+            });
+          }, {
+            // errorNotification: false TODO: DataService.list() does not support
+          });
+      };
+
+      var createBinding = function(roleName, newSubject, projectName) {
+        RoleBindingsService
```

Comment:
if i add a role to a user and the user already has that role it should not go create another role binding.  similarly if we happen to get in a situation where someone has gone and manually created multiple role bindings for the same user and role, that should get collapsed visually in the UI so that you only see it once.